### ISSUE TITLE
IAM policy on lambda execution role contains incorrect resource prefixes for writing logs when manually specifying function name

### ISSUE THREAD

**Issue by weeniearms** [serverless/serverless#6236]
# This is a Bug Report

## Description

The policies for `logs:CreateLogStream` and `logs:PutLogEvents` actions have resources with an incorrect prefix compared to the created log groups when additionally specifying the function name manually.
As of 1.45.0 these statements are generated with `arn:aws:logs:region:account:log-group:/aws/lambda/{service}-{stage}-{function}:*`, while in 1.44.1 it used to be `arn:aws:logs:region:account:log-group:/aws/lambda/{functionObject.name}:*`.

Due to this bug, no logs show up in CloudWatch for such functions as the created log groups do not match the IAM statements.

Sample config:
```
service: ${self:custom.service.name}

custom:
  service:
    name: log-group-issue-lambda

provider:
  name: aws
  runtime: nodejs10.x
  stage: ${opt:stage, 'dev'}
  region: eu-west-1
  stackName: ${self:custom.service.name}
  apiKeys:
    - ${self:custom.service.name}-key

functions:
  hello:
    name: ${self:custom.service.name}-hello
    handler: handler.hello
```

## Additional Data

* ***Serverless Framework Version you're using***: 1.45.0
* ***Operating System***: Linux 4.15.0-50-generic
* ***Stack Trace***: n/a
* ***Provider Error messages***: none

**Comment by medikoo** [comment]
@weeniearms thanks for report. 

Are you sure it's the case? Technically what changed is that following:
```
arn:aws:logs:region:account:log-group:/aws/lambda/service-stage-function:*
```
was replaced with:
```
arn:aws:logs:region:account:log-group:/aws/lambda/service-stage*:*
```

And as we tested it didn't affect configuration of CloudWatch logs.

Maybe it's some plugin that doesn't work well with a change?

**Comment by weeniearms** [comment]
Just updated the bug description with a sample config that can be used to reproduce this issue. Apparently this has to do with specifying the function name manually as it affects the name of the created log group.

**Comment by medikoo** [comment]
@weeniearms indeed, thanks for clarification, I'm looking into it now

**Comment by rdsedmundo** [comment]
I'm sorry my changes caused that. I didn't think about the `name` parameter.

But by the way, after this got merged I was wondering why we couldn't just attach the `AWSLambdaBasicExecutionRole` to all the Lambdas, which already includes all those CW Logs permissions to all the resources?

When we're using a VPC we do attach `AWSLambdaVPCAccessExecutionRole` (which also contain the CW Logs permissions) so I don't think that would be a problem. Is that right? If yes, we can remove all this code and just use AWS's Managed Policy.

**Comment by rdsedmundo** [comment]
@medikoo you closed this immediately after I posted. What do you think about my suggestion? I'm happy to create a separated issue to discuss that.

**Comment by medikoo** [comment]
@rdsedmundo sorry, it was autoclosed after https://github.com/serverless/serverless/pull/6240 got merged (patch is now released with v1.45.1).

I take `AWSLambdaBasicExecutionRole` is something that AWS setups automatically.
Are all accounts ensured to have that role? Isn't it provided only if user generates it after some AWS suggestion?

In either case, having this as a fallback sounds as good idea, and yes it deserves as separate issue.
